src/firebase_functions/pubsub.py

```diff
@@ -1,28 +1,26 @@
 '''Pub/sub trigger for the function to be triggered by Pub/Sub.'''
 
-import datetime as dt
-import functools
+import json
 import os
 import flask
+import base64
+import functools
+import datetime as dt
 
 from dataclasses import dataclass
-from typing import Any, Callable, Dict, Generic, List, TypeVar, TypedDict, Union
+from typing import Any, Callable, Generic, List, TypeVar, Union, Optional
 
 from firebase_functions.options import PubSubOptions, Sentinel, VpcOptions, Memory, IngressSettings
 from firebase_functions.manifest import EventTrigger, ManifestEndpoint
 from firebase_functions.params import BoolParam, SecretParam, StringParam, IntParam
 
 T = TypeVar('T')
 
-Request = flask.Request
-Response = flask.Response
 
-
-@dataclass(frozen=True)
+@dataclass()
 class CloudEvent(Generic[T]):
```

Comment:
Shouldn't this this always be 
```suggestion
class CloudEvent(MessagePublishedData[T]):
```

Reply:
This was mentioned in the specs, but it makes more sense to me to make it `MessagePublishedData` instead of `Generic`

Reply:
This wouldn't work as `MessagePublishedData` is the message type (the `T`), but we don't want to inherit from it